### DISCUSSION TITLE
How to use plugins.null-ls.extraOptions to add sources?

### DISCUSSION THREAD

**Discussion by garaiza-93** [nix-community/nixvim#460] [Q&A] [answered]
## Description
Currently, this option doesn't have an example to show usage. I'm lost on how to use this option to add sources that are not yet in nixvim. According to the description, it should be an attribute set.

Looking at `plugins.null-ls.sources`, it is a bunch of strings. So, I try to use a list of strings to get what I want, but no luck. Whatever quotations I use to make Nix happy, makes Lua not-so happy in the generated init.lua file.

Here's a snippet of what ends up in init.lua using the config below:
```lua
require("null-ls").setup({["sources"] = {"require(\"null-ls\").builtins.formatting.eslint_d","require(\"null-ls\").builtins.formatting.prettier_d","require(\"null-ls\").builtins.code_actions.eslint_d","require(\"null-ls\").builtins.diagnostics.eslint_d"}})
```

## Config
```nix
plugins.null-ls = {
    enable = true;
    extraOptions = {
      sources = [
        ''require("null-ls").builtins.formatting.eslint_d''
        ''require("null-ls").builtins.formatting.prettier_d''
        ''require("null-ls").builtins.code_actions.eslint_d''
        ''require("null-ls").builtins.diagnostics.eslint_d''
      ];

    };
  };
```

**Comment by garaiza-93** [comment, marked as answer]
Got an answer from the Matrix channel. you can use the `mkRaw` helper function to insert raw Lua code.

Here's what that looks like:
```nix
{ pkgs, helpers, ... }:
{
  plugins.null-ls = {
    enable = true;
    extraOptions = {
      sources = helpers.mkRaw ''
        { require("null-ls").builtins.formatting.eslint_d,
          require("null-ls").builtins.formatting.prettier_d_slim,
          require("null-ls").builtins.code_actions.eslint_d,
          require("null-ls").builtins.diagnostics.eslint_d
        }
      '';
    };
  };
}
```

**Reply by GaetanLepage** [reply]
Indeed, this should be working fine.
However, getting a proper `extraSources` option would be an helpful addition.
-> See [this issue](https://github.com/pta2002/nixvim/issues/97) for tracking.
